### DISCUSSION TITLE
How do I use yara in manifest mode?

### DISCUSSION THREAD

**Discussion by bmanga** [microsoft/vcpkg#20462] [Q&A]
The port doesn't seem to create a yaraConfig file, so how can I find it and use it though cmake?

Thanks!

**Comment by mathisloge** [comment]
looks like yara don't export anything for cmake (no findyara nor pkg-config). 
So I think you have to write a Findyara module yourself. 

CMake has a good guide for it: https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#a-sample-find-module

**Comment by bmanga** [comment]
Is this something that could be added to the vcpkg port?

**Reply by mathisloge** [reply]
some ports are doing that and export the targets with `unoffical::<target_name>`. But I generally don't like that so much because then your project has a hard dependency on vcpkg (can't be build without it).

So the best would be to just write the find module and add it to your project (this is just my recommendation, but if you are ok with a hard dependency on vcpkg, you could also open a PR and write a cmake wrapper for it, exporting the correct targets)

**Reply by bmanga** [reply]
I see. How do I get vcpkg to tell me the location of the include/lib files for yara?
Thanks for the help!

**Reply by mathisloge** [reply]
If you follow the previously linked guide from cmake, it will usually work, if you set the vcpkg Toolchain file.

https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html
